src/elementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect } from 'react'
+import { Dropdown, Header, Message } from 'semantic-ui-react'
+import { ApplicationViewProps } from '../../types'
+
+const ApplicationView: React.FC<ApplicationViewProps> = ({
+  templateElement,
+  onUpdate,
+  initialValue,
+  isEditable,
+  currentResponse,
+  validationState,
+  onSave,
+}) => {
+  const {
+    label,
+    description,
+    placeholder,
+    options,
+    default: defaultIndex,
+  } = templateElement?.parameters
+
+  const [value, setValue] = useState(initialValue ? initialValue?.text : options[defaultIndex])
+
+  useEffect(() => {
+    onUpdate(value)
+    if (!initialValue) onSave({ text: value, optionIndex: options.indexOf(value) })
+  }, [])
+
+  function handleChange(e: any, data: any) {
+    const { value } = data
+    onUpdate(value)
```

Comment:
do we need on update if onSave is called ?

Reply:
Yes, because `onUpdate` regenerates `allResponses` based on the new value, which we need. :)

Reply:
Doesn't onSave call onUpdate equivalent ?

Reply:
Ah yes, oops didn't notice that. Have removed extra onUpdate. :)  ac83836